src/ActionList/Item.tsx

```diff
@@ -201,10 +201,8 @@ export function Item(itemProps: Partial<ItemProps> & {item?: ItemInput}): JSX.El
       if (disabled) {
         return
       }
-      if (onAction) {
-        onAction(itemProps as ItemProps, event)
-      }
       onClick?.(event)
+      onAction?.(itemProps as ItemProps, event)
```

Comment:
I like the re-ordering here.  We may want to check for event.defaultPrevented on the click as well, just to give the consumer an easy way to override the behavior